beamer/agent/agent.py

```diff
@@ -114,6 +116,12 @@ def _check_target_chain(self, target_chain: _Chain) -> None:
         if not target_chain.fill_manager.functions.allowedLps(self._config.account.address).call():
             raise RuntimeError("Agent address is not whitelisted on FillManager")
 
+    def _init_fill_mutexes(self, chains: dict[ChainId, _Chain]) -> None:
+        self._fill_mutexes: dict[tuple[ChainId, ChecksumAddress], threading.Lock] = {}
+        for chain in chains.values():
+            for chain_id, address in chain.tokens:
+                self._fill_mutexes[(chain_id, address)] = threading.Lock()
```

Comment:
I am wondering about the lifetime of these locks. You create them here, which is called from `_setup_direction`. When are these locks destroyed?

Note that an agent could be stopped and started again.

Reply:
shouldnt they be destroyed when agent stopped and reinitialized?

Reply:
do you want to keep them all the time and dont replace them?

Reply:
> do you want to keep them all the time and dont replace them?

It depends. The question is: for how long should those locks live?